docs/changelog-fragments.d/708.misc.md

```diff
@@ -0,0 +1,2 @@
+Refresh documentation regarding linting code when
+proposing new changes. -- by {user}`ssbarnea`
```

Comment:
There are a few things to think about here:
1) [consistency] It's a bit weird to have an em dash after a period. So the samples don't have it. I think this should be dropped
2) [writing style] Your sentence looks like a git commit message (imperative mood, present tense, it reads as a TODO-list item now) but it should instead be in something like the past tense, explaining the high-level changes in the experiences for the reader compared to the previous release. Imagine you describe that's happened already.
3) [clarity] I think the explanation is not very specific. It reads as "something random happened, read the code to learn what exactly". Instead, we need to extract the highlights of what changed and make them prominent. This does not need to go into internal/implementation details, just a high-level "we have affected you with this specific thing".
4) [category] I'm not entirely sure that this belongs to `misc` — could you explain why you chose this category? This sounds like a contributor-facing change for which I'd go for `internal`. It also mentions the documentation for which we have `doc`. Do you have any thoughts on this?

I've explained some of these things in a detailed manner here https://github.com/ansible/ansible-navigator/pull/641/files#r765392465 — please read it.

Reply:
My toughs about this are that we spend too much time talking and complaining about form instead of just fixing issues. Too much bureaucracy for the number of fixes we need to make.

For example I would not have bothered to include a changelog frarment for that change at all. In the end documentation is out of sync, and written in way that makes impossible to keep it in sync with development. Including some output lines from tox stdout is useless, these lines would change quickly and they provide no real value to the user.

@cidrblock WDYT?

Reply:
An inconsistent changelog is an issue, I don't enjoy having to refactor multiple fragments right before a release ​just because somebody didn't care to think about them on time.

And the output lines are useful, even when outdated. They are stripped off, for this reason, just showing mostly the beginning and the form of the output to expect.

Reply:
Updated with past-tense, no ending dot, used fo docs instead of misc.

Reply:
I think I'm glad we have change logs now.
I think I will try to make my change logs past tense
I will probably always end sentences with a period.
I will probably forget if the change log gets a period or the docstring, since the 2 seem to conflict.
I also think it's might be easier to adopt changelog fragments being added first with follow on efforts to sync their grammar and tense.

Reply:
@cidrblock that sounds reasonable. I tend to use periods only if there's more than one sentence but not before `--`. I wonder if I could lint this, it'd be easier to catch early w/o a human review.

@ssbarnea thanks for doing that, I'd like to keep the discussion going, though. I think in some cases only the PR submitter may judge properly which category is needed. We need to iterate on specificity too. I recognize that it's hard and frustrating at times. The reason it's better to do it early is that when the release time comes, it's easier to understand what's about to be released and the whole thing is closer to the final/ready state and is already high-quality by that time.